templates/boilerplate/client-qx/source/class/qxapp/wrappers/webSocket.js

```diff
@@ -0,0 +1,248 @@
+/**
+ * This class is a direct link with socketio.
+ * @asset(qxapp/*)
+ * @asset(socketio/socket.io.js)
+ * @ignore(io)
+ */
+
+qx.Class.define("qxapp.wrappers.webSocket", {
+  extend: qx.core.Object,
+
+  //Socket.io events
+  events: {
+    /** socket.io connect event */
+    "connect": "qx.event.type.Event",
+    /** socket.io connecting event */
+    "connecting": "qx.event.type.Data",
+    /** socket.io connect_failed event */
+    "connect_failed": "qx.event.type.Event",
+    /** socket.io message event */
+    "message": "qx.event.type.Data",
+    /** socket.io close event */
+    "close": "qx.event.type.Data",
+    /** socket.io disconnect event */
+    "disconnect": "qx.event.type.Event",
+    /** socket.io reconnect event */
+    "reconnect": "qx.event.type.Data",
+    /** socket.io reconnecting event */
+    "reconnecting": "qx.event.type.Data",
+    /** socket.io reconnect_failed event */
+    "reconnect_failed": "qx.event.type.Event",
+    /** socket.io error event */
+    "error": "qx.event.type.Data"
+  },
+
+  properties: {
+    libReady: {
+      nullable: false,
+      init: false,
+      check: "Boolean"
+    },
+
+    url: {
+      nullable: false,
+      init: "http://".concat(window.location.hostname),
+      check: "String"
+    },
+    /** The port used to connect */
+    port: {
+      nullable: false,
+      init: Number(window.location.port),
+      check: "Number"
+    },
+    /** The namespace (socket.io namespace), can be empty */
+    namespace: {
+      nullable: true,
+      init: "",
+      check: "String"
+    },
+    /** The socket (socket.io), can be null */
+    socket: {
+      nullable: true,
+      init: null,
+      check: "Object"
+    },
+    /** Parameter for socket.io indicating if we should reconnect or not */
+    reconnect: {
+      nullable: true,
+      init: true,
+      check: "Boolean"
+    },
+    connectTimeout: {
+      nullable: true,
+      init: 10000,
+      check: "Number"
+    },
+    /** Reconnection delay for socket.io. */
+    reconnectionDelay: {
+      nullable: false,
+      init: 500,
+      check: "Number"
+    },
+    /** Max reconnection attemps */
+    maxReconnectionAttemps: {
+      nullable: false,
+      init: 1000,
+      check: "Number"
+    }
+  },
+
+  /** Constructor
+   *
+   * @param namespace {string ? null} The namespace to connect on
+   */
+  construct: function(namespace) {
+    this.base(arguments);
+    // if (namespace !== null) {
+    if (namespace) {
+      this.setNamespace(namespace);
+    }
+    this.__name = [];
+  },
+
+  members: {
+    //The name store an array of events
+    __name: null,
+
+    /**
+     * Trying to using socket.io to connect and plug every event from socket.io to qooxdoo one
+     */
+    connect: function() {
+
+      // initialize the script loading
+      var socket_io_path = "../resource/socketio/socket.io.js";
+      var dynLoader = new qx.util.DynamicScriptLoader([
+        socket_io_path
+      ]);
+
```

Comment:
since socket.io.js is so simple, I would consider adding it to the end of this file and thus load it automatically like it is done in https://github.com/oetiker/QxD3/blob/master/source/class/qxd3/Svg.js

Reply:
Out of curiosity, what would be the benefit of having it in the same file?

Reply:
a) it would be included in the overall transformation and optimization done in the compilation step
b) faster loading of the application as there is no extra load event for this file.